settings/src/components/first-time/setup-progress-bar.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { Icon, check } from '@wordpress/icons';
+
+export default function SetupProgressBar( { currentStepIndex, steps } ) {
+	const getCompletionPercentage = useCallback(
+		() => ( currentStepIndex / ( steps.length - 1 ) ) * 100,
+		[ currentStepIndex, steps.length ]
+	);
+
+	const getStepClass = ( index ) => {
+		if ( index === currentStepIndex ) {
+			return 'is-enabled';
+		}
+
+		if ( currentStepIndex > index ) {
+			return 'is-complete';
+		}
+
+		return 'is-disabled';
+	};
+
+	return (
+		<div className="wporg-2fa__progress-bar">
```

Comment:
We could likely [make this element accessible](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role) without much effort.